torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -271,6 +369,9 @@ std::vector<std::shared_ptr<NCCLComm>>& ProcessGroupNCCL::getNCCLComm(
 
   C10D_NCCL_CHECK(ncclGroupEnd());
 
+  // Hold the lock before modifying the cache.
+  std::lock_guard<std::mutex> lock(devNCCLCommMapLock_);
```

Comment:
If this is only for `devNCCLCommMap_`, is it OK to move `ncclStreams_` and `ncclEvents_` out of the scope?